test/config.py

```diff
@@ -20,5 +25,73 @@
     'irods_zone': 'tempZone',
     'irods_server_hostname': 'localhost',
 
-    "run_genquery2_tests": False
+    'run_genquery2_tests': False
+}
+
+schema = {
+    '$schema': 'http://json-schema.org/draft-07/schema#',
+    '$id': 'https://schemas.irods.org/irods-http-api/test/0.2.0/test-schema.json',
+    'type': 'object',
+    'properties': {
+        'host': {
+            'type': 'string'
+        },
+        'port': {
+            'type': 'number'
+        },
+        'url_base': {
+            'type': 'string'
+        },
+        'openid_connect': {
+            'type': 'object',
+            'properties': {
+                'mode': {
+                    'enum': [ 'client', 'protected_resource' ]
+                }
+            },
+            'required': [ 'mode' ]
+        },
+        'rodsadmin': {
+            '$ref': '#/definitions/login'
+        },
+        'rodsuser': {
+            '$ref': '#/definitions/login'
+        },
+        'irods_zone': {
+            'type': 'string'
+        },
+        'irods_server_hostname': {
+            'type': 'string'
+        },
+        'run_genquery2_tests': {
+            'type': 'boolean'
+        }
+    },
+    'required': [
+        'host',
+        'port',
+        'url_base',
+        'openid_connect',
+        'rodsadmin',
+        'rodsuser',
+        'irods_zone',
+        'irods_server_hostname',
+        'run_genquery2_tests'
+    ],
+    'definitions': {
+        'login': {
+            'type': 'object',
+            'properties': {
+                'username': {
+                    'type': 'string'
+                },
+                'password': {
+                    'type': 'string'
+                }
+            },
+            'required': [ 'username', 'password' ]
+        }
+    }
 }
+
+validate(instance=test_config, schema=schema)
```

Comment:
Can `validate` collide with things in the test python scripts?
Is it safer to qualify it with `jsonschema`?

Reply:
Within the import/config.py itself, no. Does `validate(...)` get out? Yes, if you use `config.validate(...)` after an  `import config`, or one of the other various imports to pull in validate (e.g., `from config import validate`). I think it may collide if you do a `from config import *`, but that's bad practice anywho.

Reply:
So if we used the following, that would remove the possibility of name collisions, right?
```python
import jsonschema
# ... other code ...
jsonschema.validate(args...)
```

Reply:
It turns into `config.jsonschema.validate`. Can change it to `import jsonschema` if you want, no real preferences here.

Reply:
Hmm, so we can't keep the call to `validate` private to config.py?

If that's not possible, then what you have is fine.

Reply:
I think that would require some module magic/organization that I'm not too familiar with. Maybe when the config grows bigger it could be something we look into.

Reply:
Agreed. Not a blocker right now.